src/wrapper_template.h

```diff
@@ -18,6 +18,24 @@ namespace nb = nanobind;
 // Define the type for the function to be wrapped.
 using Func = std::function<double(double)>;
 using MultiargumentFunc = std::function<double(const std::vector<std::variant<double, int>>&)>;
+inline bool check_int_dtype(const nb::object& array) {
+  return nb::isinstance<nb::ndarray<int8_t>>(array) || nb::isinstance<nb::ndarray<uint8_t>>(array) ||
+         nb::isinstance<nb::ndarray<int16_t>>(array) || nb::isinstance<nb::ndarray<uint16_t>>(array) ||
+         nb::isinstance<nb::ndarray<int32_t>>(array) || nb::isinstance<nb::ndarray<uint32_t>>(array) ||
+         nb::isinstance<nb::ndarray<int64_t>>(array) || nb::isinstance<nb::ndarray<uint64_t>>(array);
+}
+
+// Function to check whether given ndarray is c_contigous (row-major contigous)
+template <typename T>
+inline bool is_c_contiguous(const nb::ndarray<T>& arr) {
+  if (arr.size() == 0) return true;
+  size_t expected_stride = 1;
+  for (ssize_t i = arr.ndim() - 1; i >= 0; --i) {
+    if (arr.stride(i) != expected_stride) return false;
+    expected_stride *= arr.shape(i);
+  }
+  return true;
```

Comment:
Code reader may not be familiar how strides are related to C-contiguous stuff. Can you explain that in comment ?